### PR TITLE
Publication to Maven Central

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
Fixes https://github.com/Kotlin/kotlinx-datetime/issues/98
Fixes https://github.com/Kotlin/kotlinx-datetime/issues/61
Fixes https://github.com/Kotlin/kotlinx-datetime/issues/40

The TeamCity config is mostly unchanged from the autogenerated version, with one notable difference: mentions of bintray are scrubbed as to avoid clutter in the TeamCity build parameter lists.

### CODE REVIEW

**Comment by ilya-g** [comment]
It's not necessary to remove bintray parameters manually, we'll update infra and regenerate TC config again.

File: README.md [hunk 1]
```diff
@@ -254,14 +254,6 @@ and enable [core library desugaring](https://developer.android.com/studio/write/
 
 ### Gradle
 
-- Add the bintray repository:
-
-```kotlin
-repositories {
-    maven(url = "https://kotlin.bintray.com/kotlinx/") // soon will be just jcenter()
-}
```
Comment (ilya-g) [inline README.md:262]: In Gradle builds, Maven Central repository has to be defined explicitly as any other respository. So it makes sense to note "Add `mavenCentral()` to `repositories` if it's not already there, and give an example.

Comment (dkhalanskyjb) [reply README.md]: Fixed.

**Comment by dkhalanskyjb** [comment]
Sure, but is there any harm in removing them manually for now?

**Comment by ilya-g** [comment]
If not strictly necessary, please avoid changing generated code to have less conflicts there on the next update.

**Comment by dkhalanskyjb** [comment]
What conflicts do you mean? Judging by this https://github.com/Kotlin/kotlinx.team.infra/blob/master/main/src/kotlinx/team/infra/TeamCity.kt#L42-L56, the new version simply overwrites the old one without attempting any smart conflict resolution.

**Comment by dkhalanskyjb** [comment]
I see, this does make sense. I reverted the extraneous edits.

**Review by ilya-g — APPROVED** [review]

**Pull request merged**